src/Stratis.Bitcoin/Consensus/ConsensusManger.cs

```diff
@@ -74,13 +117,388 @@ public class ConsensusManager
                 utxoHash = await this.consensusRules.RewindAsync().ConfigureAwait(false);
             }
 
-            this.concurrentChain.SetTip(this.Tip);
+            lock (this.treeLock)
+            {
+                this.chainedHeaderTree.Initialize(this.Tip, !blockStoreDisabled);
+            }
+
+            this.logger.LogTrace("(-)");
+        }
+
+        /// <summary>
+        /// A list of headers are presented from a given peer,
+        /// we'll attempt to connect the headers to the tree and if new headers are found they will be queued for download.
+        /// </summary>
+        /// <param name="peerId">The peer that is providing the headers.</param>
+        /// <param name="headers">The list of new headers.</param>
+        /// <returns>The last chained header that is connected to the tree.</returns>
```

Comment:
When null?

Reply:
Actually we cannot have null, the code below is misleading.

Reply:
This is from the tree
```
            if (connectNewHeadersResult == null)
                connectNewHeadersResult = new ConnectNewHeadersResult() { Consumed = latestNewHeader };
```

Reply:
Ah you mean `newHeaders` cant be null yeah good call